### ISSUE TITLE
Donuts do not update in Frontend for images on board

### ISSUE THREAD

**Issue by leriomaggio** [DynamicGenetics/The-Learning-Machine#6]
The donuts do not change nor updates after image annotation, despite values from backend are passed to frontend changed.

This is particularly easy to spot if default number of images is decreased (e.g. to `2` ) and a non-retrained model is used (as it's the case of having heavy shifts in performance)

After a quick check, I noticed that the `refreshNodeArray` function in `d8r.js` updates links from server response, but doesn't do anything with donuts.  

Isn't there were the donut update should be performed ?

**Comment by leriomaggio** [comment]
@OliverDavis Ok so the issue with the donuts not updating lies in the `restart` function:

when updating `node.data(data.nodes, ...`) the corresponding `donut` function gets triggered only for **new** nodes added to the board (i.e. the single new node returned in response to `annotate`).

Other nodes are left unchanged, as they're still part of the `node.data`.

I believe a solution could be working on the `update` function as well 
(For what I could understand, in our current implementation only `enter` is used?).

I've tried a few solutions on my own, but with very poor results, unfortunately. 
Nonetheless, I've found this link: https://jonsadka.com/blog/how-to-create-adaptive-pie-charts-with-transitions-in-d3
which I think may be useful to rework our solution?
